cli/core/docs/design/README.md

```diff
@@ -0,0 +1,10 @@
+# Design Proposals
+
+This directory is to be populated with design reviews.
+
+Use the [Proposal Template](docs/cli/dev/_proposal.md) as a starting point.
```

Comment:
We should probably copy that file into `cli/core` as well to keep this link working

Reply:
Updated the link. Current plan is to still share the `dev` under `cli/core/docs` as it has docs like `build.md` which are still valid for cli/core.